llama_index/graph_stores/nebulagraph.py

```diff
@@ -322,25 +325,34 @@ def get_flat_rel_map(
                 f"[rel IN e | "
                 f"  [rel.`{self._rel_prop_names[0]}`, dst(rel)] "
```

Comment:
I have a question
When I use the keyword `Berkeley`
the original ngql：
> [rel.`{self._rel_prop_names[0]}`, dst(rel)]

the result of flattened_rels is.

> Berkeley, founded in, Berkeley, founded in, 1982

Is the subject information missing in this result?
Does this result lack subject information and cause ambiguity, making chatgpt think that Berkeley was founded in 1982, but it is actually Philz founded in 1982
if use 

> [src(rel), rel.`{self._rel_prop_names[0]}`, dst(rel)]

the result of flattened_rels is

> Berkeley, Philz, founded in, Berkeley, Philz, founded in, 1982

Is it possible to modify the match statement to be directed, or change the format of rel_initial_text

Reply:
Dear @BleakStone 

Thanks!

What do you think if we add a WHERE filter to ensure there is no starting node(the first subject) in the whole rel_map?

Thus we don't have `a-b-a-c` which seems confusing.

Also, it seems the relationship that we extracted isn't good enough here, we could instead leverage other extraction function as [here](https://github.com/jerryjliu/llama_index/pull/7068) introduced, and @logan-markewich provided an example on this: https://colab.research.google.com/drive/1G6pcR0pXvSkdMQlAK_P-IrYgo-_staxd?usp=sharing.

Reply:
Whether the knowledge sequence can be changed to a complete triplet
such as:
The following are knowledge sequence in max depth 2 in the form of `subject,predicate, object, subject_next_hop, predicate_next_hop, object_next_hop ...`

awesome，I've tried Babelscape/rebel-large to extract triples, and it's fast and token-saving.
At the same time, I am also trying other Chinese knowledge extraction tools, such as DeepKE

Reply:
> @wey-gu thanks for your reply
> 
> Whether the knowledge sequence can be changed to a complete triplet such as: The following are knowledge sequence in max depth 2 in the form of `subject,predicate, object, subject_next_hop, predicate_next_hop, object_next_hop ...`
> 
> awesome，I've tried Babelscape/rebel-large to extract triples, and it's fast and token-saving. At the same time, I am also trying other Chinese knowledge extraction tools, such as DeepKE

Thanks to you @BleakStone 

In such format, the subject_next_hop is actually the object or?

> At the same time, I am also trying other Chinese knowledge extraction tools, such as DeepKE

Awesome!! I am not familiar with NER/ChineseNER actually, looking forward to your exploration of this domain!